### ISSUE TITLE
Put DAX Studio in the Microsoft Store

### ISSUE THREAD

**Issue by DubbySlashy** [DaxStudio/DaxStudio#109]
This would allow DAX Studio to be automatically updated and supports Microsoft's goal of users using Microsoft Store apps over traditional programs.

**Comment by dgosbell** [comment]
It does not look like this would be too hard to do. Although a store version might only work in the stand alone mode. As far as I can see I don't think you can deploy office add-ins as a store app. This will probably require a bit more research, before we can figure out how best to implement this

**Comment by Styxxy** [comment]
Any news on this? It would nice to have it also in the Windows Store.

@dgosbell Regarding the office add-in, yes, it is possible. Docs: https://docs.microsoft.com/en-us/office/dev/store/use-the-seller-dashboard-to-submit-to-the-office-store

**Comment by dgosbell** [comment]
No news yet. I've been continuing to do background research on this and I need to alter some of the internal architecture so that we can support Windows Store, a portable version and a chocolatey package. I'm currently looking at doing some improvements to the data export user interface, but then I'm thinking of possibly tackling these "deployment options" as the next major piece of work after that.

In relation to the Office add-ins unfortunately there are multiple different types of addins on only a subset of them are supported on the Office Store (AppSource). In order to do the integration with PowerPivot the DAX Studio add-in is written using VSTO and as noted on this page  [https://docs.microsoft.com/en-us/office/dev/store/submit-to-the-office-store](https://docs.microsoft.com/en-us/office/dev/store/submit-to-the-office-store) - "Office VSTO add-ins and COM add-ins cannot be submitted to AppSource." And Windows Store apps are deliberately sandboxed off from the other apps on your system, so there are no work arounds for us to manually activate/install the add-in.

**Comment by Styxxy** [comment]
Thanks for the update. Regarding the office add-in, I could live without it being distributed through the store. It is a shame it wouldn't work, but nothing we can do about it. Might be an idea to provide a seperate installer for it?

**Comment by dgosbell** [comment]
>  It is a shame it wouldn't work, but nothing we can do about it. 
It is a shame, but it's also one of the design goals of store apps - to keep apps deliberately isolated from one another. It means that a DAX Studio store app cannot interfere with the stability of Excel, but it also means that we can't integrate deep enough to be able to talk to a PowerPivot model

> Might be an idea to provide a seperate installer for it?
While that would be technically possible, at that point we'd loose the benefit of the auto-updating store app as people would have to remember to manually update the Excel add-in. And it could lead to nasty bugs where people are running mismatched versions of the add-in and the User Interface components. So I'd be hesitant to go down this path. I think if people want the Excel integration they may need to stay with the traditional installer.

**Comment by MarkGoode53** [comment]
Hi. Any update on this? Would be very useful from a business point of view. Thanks.

**Comment by dgosbell** [comment]
I have been looking into this and other deployment options recently. I have a very early build working in a Windows Store format, but it still needs more testing and I need to figure out how to script the build so that it is repeatable and figure out the deployment process.

**Comment by MarkGoode53** [comment]
Thanks for the update Darren.


Mark Goode
Director, Business Insights

[cid:image001.png@01D5A357.667C44C0]

1540 Kalamalka Lake Rd.
Vernon, BC V1T 6V2
p: +1 (250) 541-3716
c: +1 (250) 309-3333
f: +1 (250) 542-2385
KalTireMining.com

**Comment by Sush0207** [comment]
Any update on this idea?? How can we auto update DAX studio?

**Comment by dgosbell** [comment]
So I can get the core User Interface working fine as a store app. The problem is that store apps are completely sandboxed from all other apps on the system, so I cannot register the Excel add-in or integrated with the Power BI External Tools ribbon when running as a store app. And I believe most people use at least one of those two integration points which greatly reduces the value of a store app. So I've put this on the back burner for the time being. 

I've also heard that Microsoft is making some changes to the store soon which might make it easier to bundle apps so maybe I'll revisit this when those get rolled out to see if any of this has changed.

**Comment by Sush0207** [comment]
Thanks for the response,can you please let me know,if there is any other option to auto update DAX studio?

**Comment by dgosbell** [comment]
No, all autoupdate technologies have similar limitations and do not fit well with DAX Studio.

**Comment by Kenneth-Barber** [comment]
DAX Studio doesn't necessarily need to be a UWP app, but having it in the Microsoft Store would be nice so that we can update it automatically and/or more easily. DAX Studio is one of the programs for which Patch My PC Home Updater will try to find updates, but it would be nice to not need to use a 3rd-party program for an updating experience that is better than manually downloading and running the installer of the latest version of DAX Studio.

**Comment by dgosbell** [comment]
It's not UWP that is the issue, it's the integrations with Excel and the external tools ribbon for Power BI Desktop. I could maybe make a "lite" version without both of those that would install fine from the MS store. It looks like it would also be possible to have the "full" version in the MS store that includes these integrations. But it would still require admin elevation to install the updates. Which is an issue for many users in corporate environments where admin rights are not available.
